sql/catalyst/src/main/scala/org/apache/spark/sql/catalog/v2/CatalogManager.scala

```diff
@@ -27,12 +27,18 @@ import org.apache.spark.sql.internal.SQLConf
  * A thread-safe manager for [[CatalogPlugin]]s. It tracks all the registered catalogs, and allow
  * the caller to look up a catalog by name.
  */
-class CatalogManager(conf: SQLConf) extends Logging {
+class CatalogManager(conf: SQLConf, sessionCatalog: TableCatalog) extends Logging {
```

Comment:
The flow in CatalogManager now seems a lot more confusing to me.
We now have a "built-in" session catalog, but none of the v2SessionCatalog methods (in an obvious manner) return that catalog.

Is the `sessionCatalog` in the Analyzer even match the `catalogManager.v2SessionCatalog` now, now that the configuration is optional?